fixtures/blocks/src/server/ProfilePage.block.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+/* eslint-disable import/first */
+
+import * as React from 'react';
+import {block, Suspense} from 'react';
+
+// Server
+
+import {fetch} from 'react-data/fetch';
+import loadProfileTimeline from './ProfileTimeline.block';
+
+function load(userId) {
+  return {
+    user: fetch(`/users/${userId}`).json(),
+    ProfileTimeline: loadProfileTimeline(userId),
+  };
+}
+
+// Client
+
+function ProfilePage(props, data) {
```

Comment:
I don’t think these should even be Blocks. Seems like Server Components would do?

Reply:
This Block is loading its own data. Who knows if you're going to need to use client stuff later.

This part is a bit awkward because you could just return this from the load function. Making this render more or less an identity function. We probably should have the compiler identify that pattern and return the same function for all identity ones.

However, it's not necessarily better than client if this is a frequently repeated component.

I'm not sure how you pick between a Block and a server component since you kind of lock yourself in.

Reply:
>I'm not sure how you pick between a Block and a server component since you kind of lock yourself in.

Yeah. Another thing I just realized messes with my head is that Server Components are more like `load` functions than like `Render` functions. But their naming is like `Render` functions.

Reply:
Load functions that return JSX directly, and is exported without a block wrapper, is almost like a Block except you can't pass the result to a JSX component.